lavfmuxer: support codecpar and prevent core dumps because of uninitialized parameters

cypline opened the pull request:
Fix for dvbcut to support the new introduced codecpar as replacement of codec. Newer versions of ffmpeg now rely on them. If not provided export produces coredumps.

cypline committed:
Update lavfmuxer.cpp

Fix for dvbcut to support the new introduced codecpar as replacement of codec. Newer versions of ffmpeg now rely on them. If not provided export produces coredumps.

bernhardu commented on src/lavfmuxer.cpp:
```diff
@@ -61,6 +61,15 @@ lavfmuxer::lavfmuxer(const char *format, uint32_t audiostreammask, mpgfile &mpg,
 
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(57, 14, 0)
+  // Use Parameters instead of Codec, necessary to get correct results, prevent crash and correct display of the output of av_dump
+  int ret = avcodec_parameters_from_context(s->codecpar, codec);
```
Just a nitpic, maybe we can move the declaration of ret outside the `#if`?

cypline replied:
Indeed you are right. I would propose to do the int ret=0; directly before. As I am new, shall I provide a new patch or will you just consider it in by the merge?

bernhardu replied:
Like you want - either change that commit and force push it to your branch, or I can do it.

cypline replied:
Ok, I changed both on my branch. New commit is ba1a7620dd220573e204a02cc3b3277c7a2dee92.

bernhardu commented on src/lavfmuxer.cpp:
```diff
@@ -101,6 +110,14 @@ lavfmuxer::lavfmuxer(const char *format, uint32_t audiostreammask, mpgfile &mpg,
+        av_log(NULL, AV_LOG_ERROR, "Failed to copy encoder parameters to output stream #%u\n", s->id);
+        return;
+      }
+#endif      
```
And this even more cosmetic - could you remove the trailing whitespaces?

cypline replied:
I confirm with you to have a common spacing but I seem to be just blind to see, the correct one to be use.

bernhardu replied:
Line 120 contains `#endif      `, it just did light up in `git gui` on my side.

cypline replied:
Thanks for the hint.

cypline committed:
lavfmuxer: fix usage of ret variable and cosmetic change in spacing

- Variable ret was initialized inside a ifdef clause and therefore would not compile any longer if used by code in a lower ffmpeg version.
- Remove whitespaces after endif

bernhardu commented:
Thanks, I pushed your changes as a single commit, with a few replacements of tabs to spaces.
